test/date_types.test.js

```diff
@@ -0,0 +1,251 @@
+    date = new Date();
+    date.setMilliseconds(0);
+    DateModel.create({
+      datetimeField: date,
+      timestampField: date,
+      timestampDefaultField: null,
+      dateField: dateOnly,
+    }, function(err, found) {
+      assert.ok(!err);
+      done();
+    });
+  });
+
+  it('should get model instance', function(done) {
+    DateModel.findOne({
+      where: {
+        id: 1,
+      },
+    }, function(err, found) {
+      assert.ok(!err);
+      assert.equal(found.datetimeField.toISOString(), date.toISOString());
+      assert.equal(found.timestampField.toISOString(), date.toISOString());
+      assert.equal(found.dateField.toISOString(), dateOnly.toISOString());
+      done();
+    });
+  });
+  it('timestampField shoud equal DEFAULT CURRENT_TIMESTAMP field', function(done) {
+    DateModel.findOne({
+      where: {
+        id: 1,
+      },
+    }, function(err, found) {
+      assert.ok(!err);
+      assert.equal(found.timestampField.toISOString(), found.timestampDefaultField.toISOString());
+      done();
+    });
+  });
+  it('should find model instance by datetime field', function(done) {
+    DateModel.findOne({
+      where: {
+        datetimeField: date,
+      },
+    }, function(err, found) {
+      assert.ok(!err);
+      assert.ok(found);
+      assert.equal(found.id, 1);
+      done();
+    });
+  });
+  it('should find model instance by timestamp field', function(done) {
+    DateModel.findOne({
+      where: {
+        timestampField: date,
+      },
+    }, function(err, found) {
+      assert.ok(!err);
+      assert.ok(found);
+      assert.equal(found.id, 1);
+      done();
+    });
+  });
+  it('should find model instance by date field', function(done) {
+    DateModel.findOne({
+      where: {
+        dateField: dateOnly,
+      },
+    }, function(err, found) {
+      assert.ok(!err);
+      assert.ok(found);
+      assert.equal(found.id, 1);
+      done();
+    });
+  });
+  it('set timezone to UTC +00:00', function(done) {
+    query("SET @@session.time_zone = '+00:00'", function(err) {
+      assert.ok(!err);
+      done();
+    });
+  });
+  it('now datetime and timestamp field should be different in 330 minutes 5:30 - 0:00', function(done) {
+    DateModel.findOne({
+      where: {
+        id: 1,
+      },
+    }, function(err, found) {
+      assert.ok(!err);
+      var diff = (found.datetimeField.getTime() - found.timestampField.getTime()) / 60000;
+      assert.equal(diff, 330);
+      done();
+    });
+  });
+
+  it('should disconnect when done', function(done) {
+    db.disconnect();
+    done();
+  }); });
+
+var prepareModel = function(tz, migrate, done) {
+  db = getSchema({timezone: tz});
+  DateModel = db.define('DateModel', {
+    id: {type: Number, id: 1, generated: true},
+    datetimeField: {type: Date, dataType: 'datetime', null: false},
+    timestampField: {type: Date, dataType: 'timestamp', null: false},
+    timestampDefaultField: {type: Date, dataType: 'timestamp', null: false},
```

Comment:
https://github.com/strongloop/loopback-connector-mysql/blob/master/README.md#loopback-to-mysql-types

>Type mappings:
>LoopBack to MySQL types
>Date -> DATETIME

>MySQL to LoopBack types
>DATE,TIMESTAMP,DATETIME  -> Date

I just want to make this mapping correct. Try my simple and clear test cases with current version of connector to see what is wrong.